### ISSUE TITLE
ready.jstree event fires too soon with checkbox plugin

### ISSUE THREAD

**Issue by andsav** [vakata/jstree#1847]
I am using the checkbox plugin and trying to find all nodes with "undetermined" state once the tree is fully loaded. This is the code I am using:

```
$tree.jstree({
    'core': {
        "data": data,
    },
    'plugins': ['checkbox']
}).on('ready.jstree', function () {
    var undetermined = $tree.find('.jstree-undetermined').map(function () {
        return $(this).closest('li').attr('id');
    }).get(); // []
});
```

However, undetermined contains an empty array. Setting a timeout gives the expected result:

```
$tree.jstree({
    'core': {
        "data": data,
    },
    'plugins': ['checkbox']
}).on('ready.jstree', function () {
    setTimeout(function() {
        var undetermined = groupTree.find('.jstree-undetermined').map(function () {
            return $(this).closest('li').attr('id');
        }).get();  // ["1", "2", ...]
    }, 500); 
});
```
I believe this is because the ready.jstree event fires too soon: at that point the tree is not completely built, or the undetermined class is not yet in the DOM.

**Comment by vakata** [comment]
Yes, this is normal, as such queries should not be performed using the DOM. 

The checkbox plugin itself uses the ready event to update the undetermined nodes. You'd be better off querying for ```get_top_checked```, iterating through that collection, and getting the parents of each node (if there are any) and extracting the unique values from the resulting array - those are the nodes in undetermined state. I will provide a helper method for that in the near future.

**Comment by andsav** [comment]
Thank you, I get it.  I assumed every element should be on the DOM when ready.jstree fires. The documentation says ready.jstree is triggered "after all nodes are finished loading", so I think it is a bit ambiguous in this regard. 

Yes, it would be useful to have a helper `get_undetermined`. Querying the DOM is very tempting as it is not immediately obvious how to calculate these notes. I implemented this function, as I had to use it in my project and it works perfectly. I sent you a pull request
